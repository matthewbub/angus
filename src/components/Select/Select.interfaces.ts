export interface BaseProps {
  name: string
  options?: any[]
  rest?: any
  placeholder: string
}

export interface FieldSelectProps extends BaseProps {
  ariaLabel: string
}
export interface SelectProps extends BaseProps {
  label?: string
  className?: string
  error?: string
}
