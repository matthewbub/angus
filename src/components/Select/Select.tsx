import React, { FC, forwardRef } from 'react'
import clsx from 'clsx'
import { FieldWrapper } from '../FieldWrapper'
import { FieldError } from '../FieldError'
import { FieldSelectProps, SelectProps } from './Select.interfaces'

interface ForwardedSelectProps extends SelectProps {
  ref?: React.Ref<HTMLSelectElement>;
}

const FieldSelect: FC<FieldSelectProps> = forwardRef<HTMLSelectElement, FieldSelectProps>(({
  name,
  options = [],
  placeholder,
  ariaLabel,
  ...rest
}, ref) => {
  return (
    <select
      name={name}
      ref={ref}
      placeholder={placeholder}
      className={clsx(
        'txt1 bg2',
        'placeholder:text-neutral-500/80',
        'ring-1 ring-inset ring-white/20',
        'focus:ring-2 focus:ring-inset focus:ring-teal-600',
        'block w-full rounded-md border-0 py-1.5 shadow-sm',
        'sm:text-sm sm:leading-6'
      )}
      aria-label={ariaLabel}
      {...rest}
    >
      {options && options.length > 0 && options.map((option) => (
        <option key={option.id} value={option.id}>{option.name}</option>
      ))}
    </select>
  )
});

const Select = forwardRef<HTMLSelectElement, ForwardedSelectProps>(({
  name,
  placeholder,
  register,
  options = [],
  label = '',
  className = '',
  error,
  ...rest
}, ref) => {
  return (
    <FieldWrapper
      name={name}
      label={label}
      className={className}
    >
      <FieldSelect
        name={name}
        placeholder={placeholder}
        register={register}
        options={options}
        ariaLabel={name}
        ref={ref}
        {...rest}
      />
      <FieldError
        error={error}
      />
    </FieldWrapper>
  )
});

export default Select;
