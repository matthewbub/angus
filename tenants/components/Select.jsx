import clsx from 'clsx'
import FieldWrapper from './FieldWrapper'

/**
 * Select component for select fields.
 *
 * @param {Object} props - The component props.
 * @param {string} props.name - The name attribute for the select field.
 * @param {string} props.placeholder - The placeholder text for the select field.
 * @param {Function} props.register - The register function from a form library to bind the select field.
 * @param {Array} [props.options=[]] - The options for the select field.
 * @param {...*} props.rest - Additional props for the select field.
 * @returns {JSX.Element} The rendered Input component.
 */
export default function Select({
  name,
  placeholder,
  register,
  options = [],
  label,
  className,
  error,
  ...rest
}) {
  return (
    <FieldWrapper name={name} label={label} className={className}>
      <select
        {...register(name)}
        placeholder={placeholder}
        className={clsx(
          "rounded-md w-full border-0 py-1.5 pl-3 pr-10 text-gray-900 ring-1 ring-inset ring-gray-300 focus:ring-2 focus:ring-indigo-600 sm:text-sm sm:leading-6"
        )}
        {...rest}
      >
        {options && options.length > 0 && options.map((option) => (
          <option key={option.id} value={option.id}>{option.name}</option>
        ))}
      </select>
      <div className='h-3'>
        {error && error.length > 0 && <p className="text-sm text-red-600">{error}</p>}
      </div>
    </FieldWrapper>
  )
}
