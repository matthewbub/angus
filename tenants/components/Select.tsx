import { FC } from 'react'
import clsx from 'clsx'
import FieldWrapper from './FieldWrapper'
import { FieldError } from './FieldError'

interface FieldSelectProps {
  name: string
  register: any
  options?: any[]
  rest?: any
  placeholder: string
}

interface SelectProps extends FieldSelectProps {
  label?: string
  className?: string
  error?: string
}


const FieldSelect: FC<FieldSelectProps> = ({
  name,
  register,
  options = [],
  placeholder,
  ...rest
}) => {
  return (
    <select
      {...register(name)}
      placeholder={placeholder}
      className={clsx(
        "rounded-md w-full border-0 py-1.5 pl-3 pr-10 text-gray-900 ring-1 ring-inset ring-gray-300 focus:ring-2 focus:ring-indigo-600 sm:text-sm sm:leading-6"
      )}
      {...rest}
    >
      {options && options.length > 0 && options.map((option) => (
        <option key={option.id} value={option.id}>{option.name}</option>
      ))}
    </select>
  )
}

export const Select: FC<SelectProps> = ({
  name,
  placeholder,
  register,
  options = [],
  label = '',
  className = '',
  error,
  ...rest
}) => {
  return (
    <FieldWrapper
      name={name}
      label={label}
      className={className}
    >
      <FieldSelect
        name={name}
        placeholder={placeholder}
        register={register}
        options={options}
        {...rest}
      />
      <FieldError
        error={error}
      />
    </FieldWrapper>
  )
}
