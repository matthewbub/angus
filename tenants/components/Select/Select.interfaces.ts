export interface FieldSelectProps {
  name: string
  register: any
  options?: any[]
  rest?: any
  placeholder: string
}

export interface SelectProps extends FieldSelectProps {
  label?: string
  className?: string
  error?: string
}
