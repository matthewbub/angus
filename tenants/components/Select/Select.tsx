import { FC } from 'react'
import clsx from 'clsx'
import FieldWrapper from '../FieldWrapper'
import { FieldError } from '../FieldError'
import { FieldSelectProps, SelectProps } from './Select.interfaces'

const FieldSelect: FC<FieldSelectProps> = ({
  name,
  register,
  options = [],
  placeholder,
  ariaLabel,
  ...rest
}) => {
  return (
    <select
      name={name}
      {...register(name)}
      placeholder={placeholder}
      className={clsx(
        "rounded-md w-full border-0 py-1.5 pl-3 pr-10 text-gray-900 ring-1 ring-inset ring-gray-300 focus:ring-2 focus:ring-indigo-600 sm:text-sm sm:leading-6"
      )}
      aria-label={ariaLabel}
      {...rest}
    >
      {options && options.length > 0 && options.map((option) => (
        <option key={option.id} value={option.id}>{option.name}</option>
      ))}
    </select>
  )
}

export const Select: FC<SelectProps> = ({
  name,
  placeholder,
  register,
  options = [],
  label = '',
  className = '',
  error,
  ...rest
}) => {
  return (
    <FieldWrapper
      name={name}
      label={label}
      className={className}
    >
      <FieldSelect
        name={name}
        placeholder={placeholder}
        register={register}
        options={options}
        ariaLabel={name}
        {...rest}
      />
      <FieldError
        error={error}
      />
    </FieldWrapper>
  )
}
