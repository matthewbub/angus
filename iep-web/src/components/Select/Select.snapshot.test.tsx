import React from 'react';
import { render } from '@testing-library/react';
import Select from './Select';

it('renders Select unchanged', () => {
  const { container } = render(
    <Select
      name="test-select"
      placeholder="Select an option"
      register={() => {
        return {
          onChange: () => { },
          onBlur: () => { },
          ref: () => { },
        }
      }}
      options={[
        { id: 'option1', name: 'Option 1' },
        { id: 'option2', name: 'Option 2' },
      ]}
    />)
  expect(container).toMatchSnapshot()
})