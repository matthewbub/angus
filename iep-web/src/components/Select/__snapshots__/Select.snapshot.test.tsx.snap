// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Select unchanged 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="mt-2"
    >
      <select
        aria-label="test-select"
        class="txt1 bg2 placeholder:text-neutral-500/80 ring-1 ring-inset ring-white/20 focus:ring-2 focus:ring-inset focus:ring-teal-600 block w-full rounded-md border-0 py-1.5 shadow-sm sm:text-sm sm:leading-6"
        name="test-select"
        placeholder="Select an option"
      >
        <option
          value="option1"
        >
          Option 1
        </option>
        <option
          value="option2"
        >
          Option 2
        </option>
      </select>
      <div
        class="h-4"
      />
    </div>
  </div>
</div>
`;
